tests/test_browser.py

```diff
@@ -81,55 +81,88 @@ def test__request(httpbin):
         "Content-Type"]
 
 
-@pytest.mark.parametrize('expected_content, set_value', [
-    (b":-)", True),
-    (b"", False)
+valid_enctypes_file_submit = {"multipart/form-data": True,
+                              "application/x-www-form-urlencoded": False
+                              }
+
+default_enctype = "application/x-www-form-urlencoded"
+
+# create a temporary file for testing file upload
+file_content = b":-)"
+pic_filedescriptor, pic_path = tempfile.mkstemp()
+os.write(pic_filedescriptor, file_content)
+os.close(pic_filedescriptor)
```

Comment:
This should really go within the test function. What's the issue when you do so?

Reply:
My variable expected_content is defined before the function and there is a case where the file path is needed. The other way to do would be to redefine this entry in the function after creating the temporary file.